container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java

```diff
@@ -275,14 +281,39 @@ private HttpResponse internalServerErrorResponse(HttpRequest request, RuntimeExc
         return errorResponse(request, ErrorMessage.createInternalServerError(Exceptions.toMessageString(e)));
     }
 
-    private HttpSearchResponse handleBody(HttpRequest request) {
+
+    private HttpSearchResponse handleBody(HttpRequest request){
         // Find query profile
         String queryProfileName = request.getProperty("queryProfile");
         CompiledQueryProfile queryProfile = queryProfileRegistry.findQueryProfile(queryProfileName);
         boolean benchmarkOutput = VespaHeaders.benchmarkOutput(request);
 
         // Create query
-        Query query = new Query(request, queryProfile);
+        Query query;
```

Comment:
Could you separate out this whole thing into a separate method which returns the query? (E.g queryFromRequest(HttpRequest). This method is already too long, and that you need a separate comment for the block indicated you should.

Reply:
Yes. I created the method queryFromRequest(HttpRequest, CompiledQueryProfile), which returns the query.

Reply:
👍 feel free to merge ...